sys/include/senml/saul.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2021 Silke Hofstra
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_senml_saul SenML SAUL
+ * @ingroup     sys_senml
+ * @brief       Functionality for reading @ref drivers_saul sensors as @ref sys_senml
+ *
+ * The `senml_saul` module contains functions for reading sensors using
+ * @ref drivers_saul and converting them to @ref sys_senml_cbor.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Functionality for reading @ref drivers_saul sensors as @ref sys_senml
+ *
+ * @author      Silke Hofstra <silke@slxh.eu>
+ */
+
+#ifndef SENML_SAUL_H
+#define SENML_SAUL_H
+
+#include <stdint.h>
+#include "nanocbor/nanocbor.h"
+#include "saul_reg.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Use floats instead of decimal types when encoding SAUL measurements.
+ *
+ * If this is set to `1` the @ref phydat_t values from SAUL are converted to
+ * @ref senml_numeric_t using @ref phydat_to_senml_float.
+ * Values are converted using @ref phydat_to_senml_decimal otherwise.
+ */
+#ifndef SENML_SAUL_USE_FLOATS
+#define SENML_SAUL_USE_FLOATS (0)
+#endif
```

Comment:
Since this is a configuration can you prefix them with `CONFIG_`?
```suggestion
#ifndef CONFIG_SENML_SAUL_USE_FLOATS
#define CONFIG_SENML_SAUL_USE_FLOATS (0)
#endif
```